gateway/src/resty/resolver.lua

```diff
@@ -280,24 +280,40 @@ local function valid_answers(answers)
 end
 
 local function search_dns(self, qname, stale)
+
   local search = self.search
   local dns = self.dns
   local options = self.options
   local cache = self.cache
 
-  local answers, err
-
-  for i=1, #search do
-    local query = qname .. '.' .. search[i]
-    ngx.log(ngx.DEBUG, 'resolver query: ', qname, ' search: ', search[i], ' query: ', query)
-
+  local function get_answer(query)
+    local answers, err
     answers, err = cache:get(query, stale)
-    if valid_answers(answers) then break end
+    if valid_answers(answers) then
+      return answers, nil
+    end
 
     answers, err = dns:query(query, options)
     if valid_answers(answers) then
       cache:save(answers)
-      break
+      return answers, nil
+    end
+    return nil, err
+  end
+
+  if string.sub(qname, -1) == "." then
```

Comment:
This is not optimal. Would be better to cache the `string.sub` to a local variable. The same for `qname:sub`.

https://www.youtube.com/watch?v=FfhEdF40nhQ